src/http/content.cr

```diff
@@ -72,6 +72,10 @@ module HTTP
       bytes_read = @io.read slice[0, to_read]
       @chunk_remaining -= bytes_read
 
+      if bytes_read < to_read
```

Comment:
What? No you can't assume this! EOF is when `bytes_read == 0`. Read could always return exactly 1 byte per call and you *have* to handle that case.

Reply:
Argh.

Reply:
You only want to act when `bytes_read == 0 && @chunk_remaining > 0`.

Reply:
Yeah, that's what I had in `peek` already.